packages/app/lib/ui/widgets/note_field.dart

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+import 'package:app/ui/widgets/card.dart';
+import 'package:app/ui/widgets/card_action_button.dart';
+import 'package:app/ui/widgets/card_header.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+
+typedef OnUpdate = void Function(String?);
+
+enum InputMode { read, edit }
+
+class NoteField extends StatefulWidget {
+  final String? value;
+  final OnUpdate onUpdate;
+
+  NoteField(this.value, {super.key, required this.onUpdate});
+
+  // @override
+  State<NoteField> createState() => _NoteFieldState();
+}
+
+class _NoteFieldState extends State<NoteField> {
+  InputMode _inputMode = InputMode.read;
+  late TextEditingController _textInputController;
```

Comment:
was hoping i wouldn't need the controller for this but it's seems to be the only way to make sure the input doesn't get left with extra surrounding whitespace after submitting (the form primitives don't let you manipulate the input text without defining your own controller...)